### PR TITLE
MM-31735 - ReDisplay OpenID Connect

### PR DESCRIPTION
**Pull request by sbishel**
This reverts commit f56689c707b7e9d26e127e49cb0820ffc8a1e668.

#### Summary
Displays OpenID section in System Console.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31735

#### Screenshots
Screen Shot 2021-01-14 at 4 40 54 PM
Screen Shot 2021-01-14 at 4 40 38 PM
Screen Shot 2021-01-14 at 4 40 25 PM
Screen Shot 2021-01-14 at 4 38 07 PM

### CODE REVIEW

**Comment by sbishel** [comment]
Waiting to make sure Mobile changes are merged. 
https://github.com/mattermost/mattermost-mobile/pull/5075

**Review by hahmadia — APPROVED** [review]

**Review by wiersgallak — APPROVED** [review]

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -4117,7 +4118,11 @@ const AdminDefinition = {
                         type: Constants.SettingsTypes.TYPE_CUSTOM,
                         component: OpenIdConvert,
                         key: 'OpenIdConvert',
-                        isHidden: () => true,
+                        isHidden: it.any(
```
Comment (mkraft) [inline components/admin_console/admin_definition.jsx:4121]: Should this be like this? Namely, only hidden if *none* of them are licensed?
```suggestion
                        isHidden: it.all(
```

**Comment by furqanmlk** [comment]
Getting `OAuth DEPRECATED` option for E10 license 
image

File: components/admin_console/admin_definition.jsx (hunk 1 above)
Comment (sbishel) [reply components/admin_console/admin_definition.jsx:4124]: Removed check on Google and Office365 completely. Now only check for "OpenId" license. 
So it is hidden if it is NOT Licensed for "OpenId".
It is also hidden if it is licensed for OpenId and does NOT use LegacyOAuth.

File: components/admin_console/admin_definition.jsx [hunk 2]
```diff
@@ -4062,10 +4062,14 @@ const AdminDefinition = {
                         defaultMessage='deprecated'
                     />
                 ),
-                shouldDisplay: () => false,
+                shouldDisplay: (license) => license.IsLicensed && license.OpenId === 'true',
```
Comment (sbishel) [inline components/admin_console/admin_definition.jsx:4065]: @furqanmlk - Fixes displaying "Deprecated"

File: components/admin_console/admin_definition.jsx [hunk 3]
```diff
@@ -4062,10 +4062,14 @@ const AdminDefinition = {
...
             },
             isHidden: it.any(
                 it.not(it.licensed),
+                it.all(
+                    it.licensedForFeature('OpenId'),
+                    it.not(usesLegacyOauth),
+                ),
```
Comment (sbishel) [inline components/admin_console/admin_definition.jsx:4072]: OAuth should be hidden if it is licenced for OpenId AND does NOT use legacy oauth.

**Comment by sbishel** [comment]
@mkraft - Let me know if you want to talk through some of logic on displaying. With the negative checks it gets confusing, but I think I have it straight now and tested all 6 iterations.
3 difference licensing (TE, E10, E20)
2 configuration variances (Contains legacy OAuth, NO Legacy OAuth)

**Review by furqanmlk — APPROVED** [review]
All test cases passed ✅

**Review by mkraft — APPROVED** [review]

**Pull request merged**